### PR TITLE
feat: Apply Signature Verification for SellData and Build Tx Command-line

### PR DESCRIPTION
**Pull request by inchori**
- Recreate Pull Request #256

### CODE REVIEW

**Comment by gyuguen** [comment]
I'm sorry to have you recrete this PR.
Please refer to the previous PR and reflect the reviews. :)

File: x/market/client/cli/parse.go [hunk 1]
```diff
@@ -6,7 +6,8 @@ import (
 )
 
 const (
-	FlagDealFile = "deal-file"
+	FlagDealFile    = "deal-file"
+	ReceiptDataFile = "receipt-file"
```
Comment (youngjoon-lee) [inline x/market/client/cli/parse.go:10]: probably `DataVerificationCertificateFile`.

File: x/market/client/cli/parse.go [hunk 2]
```diff
@@ -30,3 +49,18 @@ func (input *createDealInputs) UnmarshalJSON(data []byte) error {
...
+	dec := json.NewDecoder(bytes.NewReader(data))
+	dec.DisallowUnknownFields()
+
+	if err := dec.Decode(&sellData); err != nil {
+		return err
+	}
+
+	*input = sellDataInputs(sellData)
+	return nil
+}
```
Comment (youngjoon-lee) [inline x/market/client/cli/parse.go:66]:
```suggestion
func (input *sellDataInputs) UnmarshalJSON(data []byte) error {
	var sellData sellDataInputs
	dec := json.NewDecoder(bytes.NewReader(data))
	dec.DisallowUnknownFields()

	if err := dec.Decode(&sellData); err != nil {
		return err
	}

	*input = sellData
	return nil
}
```
I think we don't need `X...` structs. Please check if my code works. If so, we can also delete the `XCreateDealInputs` above.

File: x/market/keeper/deal.go [hunk 3]
```diff
@@ -138,14 +138,36 @@ func (k Keeper) SellOwnData(ctx sdk.Context, seller sdk.AccAddress, cert types.D
...
+	if !flag {
+		return sdk.Coin{}, sdkerrors.Wrap(err, "data validator is invalid address")
+	}
+
+	// Is deal's balance insufficient
+	dealBalance := k.bankKeeper.GetBalance(ctx, dealAddress, "umed")
+	sellerBalance := k.bankKeeper.GetBalance(ctx, seller, "umed")
+	if dealBalance.IsLT(sellerBalance) {
+		return sdk.Coin{}, fmt.Errorf("deal's balance is smaller than seller's balance")
+	}
```
Comment (youngjoon-lee) [inline x/market/keeper/deal.go:169]: As I understand, this code block should be deleted. Below, you're already checking whether `isTrustedValidate(...)` and `dealBalance < pricePerData`.

File: x/market/keeper/deal.go [hunk 4]
```diff
@@ -209,3 +231,19 @@ func SetCurNumData(deal types.Deal) {
 	curNumData := deal.GetCurNumData() + 1
 	deal.CurNumData = curNumData
 }
+
+func (k Keeper) Verify(ctx sdk.Context, validatorAddr sdk.AccAddress, unSignedCert types.UnsignedDataValidationCertificate, cert types.DataValidationCertificate) (bool, error) {
```
Comment (youngjoon-lee) [inline x/market/keeper/deal.go:235]: I think you forgot about this comment: https://github.com/medibloc/panacea-core/pull/256#discussion_r788489685  :)

File: x/market/keeper/deal.go (hunk 3 above)
Comment (inchori) [reply x/market/keeper/deal.go:169]: Thanks for your feedback. I think these code blocks are created when fix base branch. 😭 I'll change it!

Comment (youngjoon-lee) [reply x/market/keeper/deal.go:169]: No worries. I guessed so. :)

File: x/market/client/cli/parse.go (hunk 2 above)
Comment (hansol-medi) [reply x/market/client/cli/parse.go:66]: Agreed

**Review by hansol-medi — APPROVED** [review]
LGTM If @youngjoon-lee's comments are resolved.

**Review by youngjoon-lee — APPROVED** [review]
you can merge this PR if my comments are resolved, without re-requesting my review.

**Pull request merged**